### PR TITLE
add: [DOC] Custom Subresources Data Provider examples

### PR DESCRIPTION
**Pull request by Snowbaha**
I added example to help user to custom the Subresources
https://github.com/api-platform/core/issues/2321#issuecomment-437357871

It would be nice to set a trace in the official doc.

### CODE REVIEW

**Comment by artandor** [comment]
Hi, i was in need of the above documentation in the last few days. What's missing for this PR to be merged ? May i contribute ?

File: core/data-providers.md [hunk 1]
```diff
@@ -126,6 +128,50 @@ services:
...
+    }
+
+    public function getSubresource(string $resourceClass, array $identifiers, array $context, string $operationName = null): iterable
+    {
+        // Retrieve the blog post collection from somewhere
+        $blogPosts = $this->subresourceDataProvider->getSubresource($resourceClass, $identifiers, $context, $operationName);
+		
+        // write your own logic
+		
+		return blogPosts;
```
Comment (vincentchalamon) [inline core/data-providers.md:159]:
```suggestion
            return blogPosts;
```

**Review by vincentchalamon — CHANGES_REQUESTED** [review]

File: core/data-providers.md [hunk 2]
```diff
@@ -126,6 +128,50 @@ services:
...
+    }
+
+    public function getSubresource(string $resourceClass, array $identifiers, array $context, string $operationName = null): iterable
+    {
+        // Retrieve the blog post collection from somewhere
+        $blogPosts = $this->subresourceDataProvider->getSubresource($resourceClass, $identifiers, $context, $operationName);
+		
+        // write your own logic
+		
+            return blogPosts;
```
Comment (vincentchalamon) [inline core/data-providers.md:159]:
```suggestion
        return blogPosts;
```

**Review by vincentchalamon — APPROVED** [review]

File: core/data-providers.md [hunk 3]
```diff
@@ -126,6 +128,50 @@ services:
...
+    public function supports(string $resourceClass, string $operationName = null, array $context = []): bool
+    {
+        return BlogPost::class === $resourceClass;
+    }
+
+    public function getSubresource(string $resourceClass, array $identifiers, array $context, string $operationName = null): iterable
+    {
+        // Retrieve the blog post collection from somewhere
+        $blogPosts = $this->subresourceDataProvider->getSubresource($resourceClass, $identifiers, $context, $operationName);
+
```
Comment (vincentchalamon) [inline core/data-providers.md:156]:
```suggestion

```

File: core/data-providers.md [hunk 4]
```diff
@@ -126,6 +128,50 @@ services:
...
+        return BlogPost::class === $resourceClass;
+    }
+
+    public function getSubresource(string $resourceClass, array $identifiers, array $context, string $operationName = null): iterable
+    {
+        // Retrieve the blog post collection from somewhere
+        $blogPosts = $this->subresourceDataProvider->getSubresource($resourceClass, $identifiers, $context, $operationName);
+		
+        // write your own logic
+
```
Comment (vincentchalamon) [inline core/data-providers.md:158]:
```suggestion

```

**Pull request merged**